spring-cloud-starter-stream-source-rabbit/src/main/java/org/springframework/cloud/stream/app/rabbit/source/RabbitSourceConfiguration.java

```diff
@@ -77,27 +82,32 @@ public MessageProperties toMessageProperties(AMQP.BasicProperties source, Envelo
 	@Autowired
 	private ConnectionFactory rabbitConnectionFactory;
 
+	private CachingConnectionFactory ownConnectionFactory;
+
 	@Bean
-	public SimpleMessageListenerContainer container() {
-		SimpleMessageListenerContainer container = new SimpleMessageListenerContainer(this.rabbitConnectionFactory);
+	public SimpleMessageListenerContainer container() throws Exception {
+		ConnectionFactory connectionFactory = this.properties.isOwnConnection()
+				? buildLocalConnectionFactory()
+				: this.rabbitConnectionFactory;
+		SimpleMessageListenerContainer container = new SimpleMessageListenerContainer(connectionFactory);
 		Listener listenerProperties = this.rabbitProperties.getListener();
-		AcknowledgeMode acknowledgeMode = listenerProperties.getAcknowledgeMode();
+		AcknowledgeMode acknowledgeMode = listenerProperties.getSimple().getAcknowledgeMode();
```

Comment:
not sure what the addition of getSimple is, but will merge and sort it out later as ATM it works for the sink case.

Reply:
This was a deprecation in prep for Boot 2.0 (we added a second container type `direct`).

I just noticed I blindly ported the code from the sink; we don't need the `bootProperties` field; we already have a `RabbitProperties` field; will push a fix.